anyio/pytest_plugin.py

```diff
@@ -58,45 +58,46 @@ def wrapper(*args, **kwargs):
     yield
 
 
+@pytest.hookimpl(trylast=True)
 def pytest_generate_tests(metafunc):
     marker = metafunc.definition.get_closest_marker('anyio')
-    if marker:
-        metafunc.fixturenames.append('anyio_backend')
-
-    if 'anyio_backend' in metafunc.fixturenames:
-        backends = metafunc.config.getoption('anyio_backends')
-        if backends == 'all':
-            backends = BACKENDS
-        else:
+    if marker or 'anyio_backend' in metafunc.fixturenames:
+        backends = marker.kwargs.get('backend') if marker else []
```

Comment:
why not "else None"?

Reply:
You're looking at an outdated commit. Another hook expected `backends` to be iterable.